node/router/src/inbound.rs

```diff
@@ -125,6 +125,9 @@ pub trait Inbound<N: Network>: Reading + Outbound<N> {
                 if !self.router().cache.contains_outbound_peer_request(peer_ip) {
                     bail!("Peer '{peer_ip}' is not following the protocol (unexpected peer response)")
                 }
+                if self.router().node_type().is_validator() && !self.router().allow_external_peers() {
```

Comment:
nit: is_validator is no longer necessary as part of the check (because allow_external_peers takes precedence)